packages/server-wallet/src/__test-with-peers__/create-and-close-channel/direct-funding.test.ts

```diff
@@ -51,59 +52,12 @@ it('Create a directly funded channel between two wallets ', async () => {
   //        A <> B
   // PreFund0
   const resultA0 = await peerWallets.a.createChannel(createChannelParams);
-
+  await messageService.send(getMessages(resultA0));
   channelId = resultA0.channelResults[0].channelId;
 
-  expect(getChannelResultFor(channelId, resultA0.channelResults)).toMatchObject({
-    status: 'opening',
-    turnNum: 0,
-  });
-
-  // Let's check what wallet a is waiting for. Expect it to be 'theirPreFundSetup' or ''
-  // (if the objective has never been cranked)
-  const waitingForBefore = (await peerWallets.a.getObjective(`OpenChannel-${channelId}`))
-    .waitingFor;
-  expect(waitingForBefore).toEqual(WaitingFor.theirPreFundSetup);
-
-  //    > PreFund0A
-  const resultB0 = await peerWallets.b.pushMessage(
-    getPayloadFor(participantB.participantId, resultA0.outbox)
-  );
-
-  expect(getChannelResultFor(channelId, resultB0.channelResults)).toMatchObject({
-    status: 'proposed',
-    turnNum: 0,
-  });
-
   //      PreFund0B
   const resultB1 = await peerWallets.b.joinChannel({channelId});
-  expect(getChannelResultFor(channelId, [resultB1.channelResult])).toMatchObject({
-    status: 'opening',
-    turnNum: 0,
-  });
-
-  //  PreFund0B <
-  const resultA1 = await peerWallets.a.pushMessage(
-    getPayloadFor(participantA.participantId, resultB1.outbox)
-  );
-
-  //  a was previously blocked on receiving b's prefund state. After pushing it in to a,
-  //  a's OpenChannel objective should have made some progress (it is now waiting on something else)
-  const waitingForAfter = (await peerWallets.a.getObjective(`OpenChannel-${channelId}`)).waitingFor;
-  expect(waitingForAfter).not.toEqual(waitingForBefore);
-  expect(waitingForAfter).toEqual(WaitingFor.funding);
-
-  /**
-   * In this case, there is no auto-advancing to the running stage. Instead we have
-   * an intermediate 'opening' stage where each party must fund their channel. A funds
-   * first, and then B funds. A and B both signs turnNum 3 on the call to updateFundingForChannels
-   * and then sends the newly signed state to each other at the same time.
-   */
-
-  expect(getChannelResultFor(channelId, resultA1.channelResults)).toMatchObject({
-    status: 'opening',
-    turnNum: 0,
-  });
```

Comment:
One drawback of the messaging services is that it becomes slightly more difficult to check the results at every step (since the messages are automatically sent). Perhaps a `pause` method could make these checks easier? Although I'm not certain it's worth the effort?

Reply:
What's the goal of the test? Is it to assert that a specific sequence of messages happens in a specific order?

If that's the goal, then the `TestMessageService` can be used to that effect:
- It can _record_ which messages are exchanged, in what order.
- We can then _expect_ a statement about the messages that are exchanged

Reply:
I think the previous checks weren't looking at the `messages` but at the `channelResult` to check if the channel is in the correct state.

If we want to inspect the `channelResult` (or `message` payloads) at each step does it make more sense to just have the test manually call `pushMessage` like before? It seems like adding a `TestMessageService` then having to scan through the history is more work then just manually handing the pushMessage results.

Reply:
Let me restate: What _should_ the goal be of the `with-peers` tests?

- Should the goal be to ensure that the correct messages are passed, when one party calls `createChannels`?
- Should the goal be to ensure that when one party calls `createChannels`, that the promise resolves? and at the end, the party has the right number of funded channels?

> the previous checks weren't looking at the messages but at the channelResult to check if the channel is in the correct state.

I don't think it's necessarily wrong to have this duplicated, but there are already "unit tests" of these crankers that assert on the `channelResult`. So, maybe this test doesn't also need to do that.

> It seems like adding a TestMessageService then having to scan through the history is more work then just manually handing the pushMessage results.

IMO it's not a question about how much work it is, it's about what the test is testing. If it's meant to be testing what messages get passed back and forth, then I think writing the `TestMessageService` _once_ to record messages that are sent is worth it. You can then use the `TestMessageService` to assert that:
- in _this_ situation, these messages are exchanged
- in _that_ situation, those messages are exchanged

Again, I do not know if that's what we're trying to test here.